lib/src/json_schema/json_schema.dart

```diff
@@ -258,14 +288,17 @@ class JsonSchema {
       _metaSchemaCompleter = _root._metaSchemaCompleter;
       _metaschemaVocabulary = metaschemaVocabulary;
       _schemaAssignments = _root._schemaAssignments;
+      _customVocabMap = _root._customVocabMap;
     }
     if (_root._isSync) {
       _validateSchemaSync();
     } else {
       if (!_root._metaSchemaCompleter.isCompleted) {
         // Wait here until the vocabularies from the metaschema have been resolved.
         // This should only need to happen once for the _root object.
-        _root._metaSchemaCompleter.future.then((_) => _validateSchemaAsync());
+        _root._metaSchemaCompleter.future
+            .then((_) => _validateSchemaAsync())
+            .onError((e, stack) => _root._thisCompleter.completeError(e, stack));
```

Comment:
This `onError` is why my tests around exceptions were failing. This moves errors from the metaSchema fetcher future into the future that is returned to users.

Reply:
:til: